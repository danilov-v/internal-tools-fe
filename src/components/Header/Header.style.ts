import styled from 'styled-components';
import { Link } from '@reach/router';

type HumburgerProps = {
  expanded: boolean;
};

export const Header = styled.header`
  position: fixed;
  top: 0;
  left: 4vw;
  z-index: 100;
  width: 92vw;
  min-height: 65px;
  padding-top: 1em;
  font-size: 1.2rem;
  border-bottom: 2px solid #616262ff;

  @media screen and (min-width: 768px) {
    display: flex;
    padding-top: 0;
    align-items: flex-end;
    padding-bottom: 1em;
  }
`;

export const NavBar = styled.ul<HumburgerProps>`
  list-style-type: none;
  padding: 0;
  display: ${({ expanded }) => (expanded ? 'block' : 'none')};
  margin-top: 3rem;

  @media screen and (min-width: 768px) {
    margin: 0;
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: baseline;
  }
`;

export const NavItem = styled.li`
  text-align: center;
  margin-left: 2.2rem;

  @media screen and (max-width: 768px) {
    margin: 0;

    &:not(:first-child) {
      margin-top: 1.2rem;
    }
  }
`;

export const NavLink = styled(Link)`
  color: #cccccc;
  text-decoration: none;
  position: relative;

  &:hover {
    color: #ffdc6282;
  }
  &[state='active'] {
    color: #ffdc62ff;

    &::before {
      content: '';
      display: block;
      position: absolute;
      top: -1rem;
      left: 40%;
      width: 12px;
      height: 12px;
      background-color: #ffdc62ff;
      clip-path: polygon(50% 100%, 0 0, 100% 0);
    }
  }
`;

export const Humburger = styled.div<HumburgerProps>`
  position: absolute;
  top: 1em;
  right: 0;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: 3rem;
  height: 3rem;
  transform: rotate(0deg);
  transition: 0.5s ease-in-out;
  cursor: pointer;

  @media screen and (min-width: 768px) {
    display: none;
  }

  div {
    height: 0.25rem;
    background: #cccccc;
    border-radius: 9px;
    opacity: 1;
    left: 0;
    transform-origin: left center;
    transition: 0.25s ease-in-out;
  }

  & div:nth-child(1) {
    transform: ${({ expanded }) =>
      expanded ? 'rotate(45deg)' : 'rotate(0deg)'};
  }
  & div:nth-child(2) {
    opacity: ${({ expanded }) => (expanded ? 0 : 1)};
  }
  & div:nth-child(3) {
    transform: ${({ expanded }) =>
      expanded ? 'rotate(-45deg)' : 'rotate(0deg)'};
  }
`;

export const UserAvatar = styled.div`
  position: absolute;
  top: 1.6em;
  right: calc(100vw - 260px);

  @media screen and (min-width: 768px) {
    right: 0;
  }
`;
