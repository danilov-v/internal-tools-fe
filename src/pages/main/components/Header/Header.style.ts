import styled from 'styled-components';
import { Link } from '@reach/router';

type HumburgerProps = {
  expanded: boolean;
};

const LINE_COLOR = '#616262';

export const Header = styled.header`
  position: sticky;
  top: 0;
  display: flex;
  justify-content: space-between;
  padding: 15px 0;
  z-index: 100;
  border-bottom: 2px solid ${LINE_COLOR};
  background: linear-gradient(90deg, #44494b 0%, #4c5153 25%, #4c5153 100%);

  @media screen and (min-width: 768px) {
    align-items: flex-end;
    justify-content: initial;
  }
`;

export const Logo = styled.div`
  cursor: pointer;
`;

export const NavBar = styled.div<HumburgerProps>`
  list-style-type: none;
  padding: 0;
  display: ${({ expanded }) => (expanded ? 'flex' : 'none')};
  flex-direction: column;
  margin-top: 80px;

  @media screen and (min-width: 768px) {
    display: block;
    flex: 1;
    margin-top: 0;
  }
`;

export const NavItem = styled.div`
  display: block;
  text-align: center;
  margin-bottom: 20px;

  @media screen and (min-width: 768px) {
    display: inline-block;
    margin-left: 40px;
    margin-bottom: 0;
  }
`;

export const NavLink = styled(Link)`
  color: #cccccc;
  text-decoration: none;
  position: relative;

  &:hover {
    color: #ffdc6282;
  }
  &[state='active'] {
    color: #ffdc62;

    &::before {
      content: '';
      display: block;
      position: absolute;
      top: -12px;
      left: calc(50% - 5px);
      width: 10px;
      height: 8px;
      background-color: #ffdc62;
      clip-path: polygon(50% 100%, 0 0, 100% 0);
    }
  }
`;

export const Humburger = styled.div<HumburgerProps>`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 45px;
  height: 35px;
  transform: rotate(0deg);
  transition: 0.5s ease-in-out;
  cursor: pointer;

  @media screen and (min-width: 768px) {
    display: none;
  }

  div {
    height: 3px;
    background: #cccccc;
    border-radius: 2px;
    opacity: 1;
    left: 0;
    transform-origin: left center;
    transition: 0.25s ease-in-out;
  }

  & div:nth-child(1) {
    transform: ${({ expanded }) =>
      expanded ? 'rotate(45deg)' : 'rotate(0deg)'};
  }
  & div:nth-child(2) {
    opacity: ${({ expanded }) => (expanded ? 0 : 1)};
  }
  & div:nth-child(3) {
    transform: ${({ expanded }) =>
      expanded ? 'rotate(-45deg)' : 'rotate(0deg)'};
  }
`;

export const UserRank = styled.div`
  display: none;

  @media screen and (min-width: 768px) {
    display: block;
  }
`;

export const Actions = styled.div`
  display: flex;
  align-items: flex-end;
  border-left: 2px solid ${LINE_COLOR};
  margin-left: 20px;
  height: 40px;
  justify-content: space-around;
  flex-basis: 120px;
`;

export const Action = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-basis: 30px;
  padding: 4px 0;
  cursor: pointer;

  &:hover {
    opacity: 0.7;
  }
`;

export const ActionIcon = styled.img`
  width: 15px;
  height: 15px;
`;
