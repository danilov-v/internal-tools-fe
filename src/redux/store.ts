import { configureStore, EnhancedStore } from '@reduxjs/toolkit';

import { appSlice, AppState } from './app/slice';
import { penaltySlice, PenaltyState } from './penalty/slice';
import { penaltyTypeSlice, PenaltyTypeState } from './penaltyType/slice';
import { personnelSlice, PersonnelState } from './personnel/slice';
import {
  personnelDetailsSlice,
  PersonnelDetailsState,
} from './personnel-details/slice';
import { positionSlice, PositionState } from './position/slice';
import { profileSlice, ProfileState } from './profile/slice';
import { promotionSlice, PromotionState } from './promotion/slice';
import { promotionTypeSlice, PromotionTypeState } from './promotionType/slice';
import {
  personnelRemovalTypeSlice,
  PersonnelRemovalTypeState,
} from './personnelRemovalType/slice';

import { rankSlice, RankState } from './rank/slice';
import { unitSlice, UnitState } from './unit/slice';
import { maritalStatusSlice, MaritalStatusState } from './maritalStatus/slice';

export const getStore = (initialState?: OptionalRootStore): EnhancedStore =>
  configureStore({
    reducer: {
      app: appSlice.reducer,
      penalty: penaltySlice.reducer,
      penaltyType: penaltyTypeSlice.reducer,
      personnel: personnelSlice.reducer,
      personnelDetails: personnelDetailsSlice.reducer,
      personnelRemovalType: personnelRemovalTypeSlice.reducer,
      position: positionSlice.reducer,
      profile: profileSlice.reducer,
      promotion: promotionSlice.reducer,
      promotionType: promotionTypeSlice.reducer,
      rank: rankSlice.reducer,
      unit: unitSlice.reducer,
      maritalStatus: maritalStatusSlice.reducer,
    },
    preloadedState: initialState,
  });

export type RootStore = {
  app: AppState;
  penalty: PenaltyState;
  penaltyType: PenaltyTypeState;
  personnel: PersonnelState;
  personnelDetails: PersonnelDetailsState;
  personnelRemovalType: PersonnelRemovalTypeState;
  position: PositionState;
  profile: ProfileState;
  promotion: PromotionState;
  promotionType: PromotionTypeState;
  rank: RankState;
  unit: UnitState;
  maritalStatus: MaritalStatusState;
};

export type OptionalRootStore = Partial<RootStore>;
