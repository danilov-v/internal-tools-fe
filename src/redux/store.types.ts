import { ProfileState } from './profile/types';
import { PersonnelState } from './personnel/types';
import { RankState } from './rank/types';
import { UnitState } from './unit/types';
import { PersonnelDetailsState } from './personnel-details/types';

export type RootStore = {
  profile: ProfileState;
  personnel: PersonnelState;
  rank: RankState;
  unit: UnitState;
  personnelDetails: PersonnelDetailsState;
};

export type OptionalRootStore = Partial<RootStore>;
