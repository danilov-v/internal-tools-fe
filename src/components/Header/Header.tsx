import React from 'react';
import { useLocalStore, useObserver } from 'mobx-react-lite';

import * as S from './Header.style';

type HeaderProps = {
  test: string;
};

const Header: React.FC<HeaderProps> = ({ test }) => {
  const counterStore = useLocalStore(() => ({
    counter: 0,
    increment() {
      counterStore.counter += 1;
    },
  }));

  return useObserver(() => (
    <S.Header>
      There will be header. <br />
      Test params - {test}! <br />
      Test store value - {counterStore.counter} <br />
      <button type="button" onClick={counterStore.increment}>
        Increment
      </button>
    </S.Header>
  ));
};

export { Header };
