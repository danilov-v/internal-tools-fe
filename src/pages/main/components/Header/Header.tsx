import React, { useState } from 'react';

import { Logo } from 'components/Logo';
import { Avatar } from 'pages/main/components/Avatar';
import * as S from './Header.style';

type IsActivePropsType = {
  isPartiallyCurrent: boolean;
};

const isActive = ({ isPartiallyCurrent }: IsActivePropsType): object => {
  return { state: isPartiallyCurrent ? 'active' : 'not-active' };
};

const testUser = {
  lastName: 'Стороженко',
  rank: 'captain',
  position: 'Командир роты',
};

const Header: React.FC<{}> = () => {
  const [isExpanded, setExpanded] = useState(false);
  const toggleExpanded = (): void => setExpanded(!isExpanded);

  return (
    <S.Header>
      <S.UserLogo>
        <Logo />
      </S.UserLogo>
      <S.NavBar expanded={isExpanded}>
        <S.NavItem>
          <S.NavLink getProps={isActive} to="/personnel">
            Рота
          </S.NavLink>
        </S.NavItem>
        <S.NavItem>
          <S.NavLink getProps={isActive} to="/schedule">
            График
          </S.NavLink>
        </S.NavItem>
      </S.NavBar>
      <S.UserAvatar>
        <Avatar user={testUser} />
      </S.UserAvatar>
      <S.Humburger expanded={isExpanded} onClick={toggleExpanded}>
        <div />
        <div />
        <div />
      </S.Humburger>
    </S.Header>
  );
};

export { Header };
