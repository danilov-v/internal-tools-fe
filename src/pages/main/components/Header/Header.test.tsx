import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { Provider } from 'react-redux';

import { getStore } from 'redux/store';

import { Header } from './Header';

describe('<Header />', () => {
  const getComponent = (): RenderResult => {
    const store = getStore();

    return render(
      <Provider store={store}>
        <Header />
      </Provider>,
    );
  };

  it('should render and match snapshot', () => {
    expect(getComponent().baseElement).toMatchSnapshot();
  });
});
