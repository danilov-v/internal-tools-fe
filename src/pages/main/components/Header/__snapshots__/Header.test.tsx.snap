// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render and match snapshot 1`] = `
<body>
  <div>
    <header
      class="sc-AxgMl fJZqfz"
    >
      <div
        class="sc-AxheI kEoNYT"
      >
        <div
          class="sc-AxjAm khWceK"
          title="logo"
        >
          <img
            alt="logo"
            class="sc-AxirZ dbNCkZ"
            src="logo.svg"
          />
        </div>
      </div>
      <div
        class="sc-Axmtr empjiD"
      >
        <div
          class="sc-AxmLO ACpjz"
        >
          <a
            class="sc-fzozJi lbnCcF"
            href="/personnel"
            state="not-active"
          >
            Рота
          </a>
        </div>
        <div
          class="sc-AxmLO ACpjz"
        >
          <a
            class="sc-fzozJi lbnCcF"
            href="/schedule"
            state="not-active"
          >
            График
          </a>
        </div>
      </div>
      <div
        class="sc-fzpans faASHH"
      />
      <div
        class="sc-fzplWN fjTdmU"
      >
        <div
          class="sc-fznyAO ivNEJa"
        >
          <img
            alt="profile image"
            class="sc-fznKkj heRQzp"
            src="profile.svg"
          />
        </div>
        <div
          class="sc-fznyAO ivNEJa"
        >
          <img
            alt="setting image"
            class="sc-fznKkj heRQzp"
            src="settings.svg"
          />
        </div>
        <div
          class="sc-fznyAO ivNEJa"
        >
          <img
            alt="logout image"
            class="sc-fznKkj heRQzp"
            src="logout.svg"
          />
        </div>
      </div>
      <div
        class="sc-fzoLsD HIAoK"
      >
        <div />
        <div />
        <div />
      </div>
    </header>
  </div>
</body>
`;
